src/System.Net.Security/tests/FunctionalTests/KerberosTest.cs

```diff
@@ -0,0 +1,331 @@
+                string target = string.Format("{0}@{1}", TestConfiguration.HostTarget, TestConfiguration.Realm);
+                NetworkCredential credential = new NetworkCredential(user.Substring(1), TestConfiguration.Password);
+                Assert.Throws<AuthenticationException>(() =>
+                {
+                    client.AuthenticateAsClientAsync(credential, target);
+                });
+            }
+        }
+
+        [Fact, OuterLoop]
+        [PlatformSpecific(PlatformID.Linux | PlatformID.OSX)]
+        public void NegotiateStream_StreamToStream_KerberosAuthInvalidPassword_Failure()
+        {
+            if (!_isKrbAvailable)
+            {
+                return;
+            }
+
+            VirtualNetwork network = new VirtualNetwork();
+
+            using (var clientStream = new VirtualNetworkStream(network, isServer: false))
+            using (var client = new UnixGssFakeNegotiateStream(clientStream))
+            {
+                Assert.False(client.IsAuthenticated);
+
+                string user = string.Format("{0}@{1}", TestConfiguration.KerberosUser, TestConfiguration.Realm);
+                string target = string.Format("{0}@{1}", TestConfiguration.HostTarget, TestConfiguration.Realm);
+                NetworkCredential credential = new NetworkCredential(user, TestConfiguration.Password.Substring(1));
+                Assert.Throws<AuthenticationException>(() =>
+                {
+                    client.AuthenticateAsClientAsync(credential, target);
+                });
+            }
+        }
+
+        [Fact, OuterLoop]
+        [PlatformSpecific(PlatformID.Linux | PlatformID.OSX)]
+        [ActiveIssue(6114, PlatformID.Linux | PlatformID.OSX)]
+        public void NegotiateStream_StreamToStream_KerberosAuthInvalidTarget_Failure()
+        {
+            if (!_isKrbAvailable)
+            {
+                return;
+            }
+
+            VirtualNetwork network = new VirtualNetwork();
+
+            using (var clientStream = new VirtualNetworkStream(network, isServer: false))
+            using (var client = new UnixGssFakeNegotiateStream(clientStream))
+            {
+                Assert.False(client.IsAuthenticated);
+
+                string user = string.Format("{0}@{1}", TestConfiguration.KerberosUser, TestConfiguration.Realm);
+                string target = string.Format("{0}@{1}", TestConfiguration.HostTarget, TestConfiguration.Realm);
+                NetworkCredential credential = new NetworkCredential(user, TestConfiguration.Password);
+                Assert.Throws<AuthenticationException>(() =>
+                {
+                    client.AuthenticateAsClientAsync(credential, target.Substring(1));
+                });
+            }
+        }
+
+        public void Dispose()
+        {
+            try
+            {
+                KDCSetup.CheckAndInitializeKerberos();
+            }
+            catch
+            {
+            }
+        }
+    }
+}
```

Comment:
I assume more tests will be coming in another round, for things like reading/writing/etc.?